executor/merge_join.go

```diff
@@ -377,7 +377,13 @@ func (e *MergeJoinExec) fetchNextInnerRows() (err error) {
 // fetchNextOuterRows fetches the next Chunk of outer table. Rows in a Chunk
 // may not all belong to the same join key, but are guaranteed to be sorted
 // according to the join key.
-func (e *MergeJoinExec) fetchNextOuterRows(ctx context.Context) (err error) {
+func (e *MergeJoinExec) fetchNextOuterRows(ctx context.Context, requiredRows int) (err error) {
+	// It's hard to calculate selectivity if there is any filter or it's inner join,
+	// so we just push the requiredRows down when it's outer join and has no filter.
+	if IsOuterJoiner(e.joiner) && len(e.outerTable.filter) == 0 {
```

Comment:
the property "is outer join" only need to be calculated once, we can use a variable to store that result

Reply:
It's updated.